modules/core/src/matrix.cpp

```diff
@@ -692,23 +692,21 @@ void Mat::create(int d, const int* _sizes, int _type)
     if( total() > 0 )
     {
         MatAllocator *a = allocator, *a0 = getDefaultAllocator();
-#ifdef HAVE_TGPU
-        if( !a || a == tegra::getAllocator() )
-            a = tegra::getAllocator(d, _sizes, _type);
-#endif
         if(!a)
             a = a0;
         try
         {
             u = a->allocate(dims, size, _type, 0, step.p, ACCESS_RW /* ignored */, USAGE_DEFAULT);
             CV_Assert(u != 0);
+            allocator = a;
```

Comment:
`umatrix.cpp` has the similar code related to Mat (besides of tons of code related to UMat): https://github.com/opencv/opencv/blob/4.10.0/modules/core/src/umatrix.cpp#L601

Reply:
UMat handles allocators a bit different. I propose to handle it with dedicated PR. Need to discuss it first.